### PR TITLE
Application Security Group Resource

### PR DESCRIPTION
**Pull request by ASHR4**
## Description

Initial start on creating the application security group resource for this AVM.

- Worked through TODO comments
- Added default example for PR Checks to pass
- Remove unsupported interfaces
- Amend supported interfaces as they were old and failing pr checks

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [X] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `locals.version.tf.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `locals.version.tf.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `locals.version.tf.json`.
  - [ ] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by ASHR4** [comment]
@microsoft-github-policy-service agree

File: outputs.tf [hunk 1]
```diff
@@ -1,11 +0,0 @@
...
-  description = "A map of private endpoints. The map key is the supplied input to var.private_endpoints. The map value is the entire azurerm_private_endpoint resource."
-  value       = azurerm_private_endpoint.this
-}
-
-# Module owners should include the full resource via a 'resource' output
-# https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr2---category-outputs---additional-terraform-outputs
-output "resource" {
-  description = "This is the full output for the resource."
-  value       = azurerm_resource_group.TODO # TODO: Replace this dummy resource azurerm_resource_group.TODO with your module resource
-}
```
Comment (CiucurDaniel) [inline outputs.tf:11]: Maybe we should keep this output according to the (docs)[https://azure.github.io/Azure-Verified-Modules/specs/terraform/#id-tffr2---category-outputs---additional-terraform-outputs] and output `azurerm_application_security_group.this`

File: variables.tf [hunk 2]
```diff
@@ -93,81 +35,23 @@ variable "location" {
...
+    kind = string
     name = optional(string, null)
-    kind = optional(string, "None")
   })
-  default     = {}
-  description = "The lock level to apply. Default is `None`. Possible values are `None`, `CanNotDelete`, and `ReadOnly`."
-  nullable    = false
+  default     = null
+  description = <<DESCRIPTION
+  Controls the Resource Lock configuration for this resource. The following properties can be specified:
```
Comment (CiucurDaniel) [inline variables.tf:43]: Maybe would be good to mention the lock object itself is `(Optional)`

**Review by CiucurDaniel — CHANGES_REQUESTED** [review]

File: outputs.tf (hunk 1 above)
Comment (ASHR4) [reply outputs.tf:11]: Hi @CiucurDaniel, I was only removing it passes the pr and pre-commit checks.

I was planning on leaving that to you so you can get hands on?

Let me know if you'd prefer me to add it :)

**Review by CiucurDaniel — APPROVED** [review]
Thanks for initial code @ASHR4!

File: variables.tf (hunk 2 above)
Comment (ASHR4) [reply variables.tf:43]: This was copied from the required interfaces: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#resource-locks

Let me know if this is still required :)

**Comment by ASHR4** [comment]
@herms14 do you mind reviewing and approving the workflows to get this initial start merged in?

**Comment by CiucurDaniel** [comment]
> Hi @CiucurDaniel,
> 
> As per the message from @herms14 the README also needs updating.
> 
> Although it's a relatively small module, Im hesitant to make all the changes as I know you are keen to contribute!
> 
> I've invited you to my forked repository so you can update this branch with the outputs you mentioned, updating the README and anything else you think will be valuable for this module.
> 
> Feel free to raise the PR or let me know and I can :)
> 
> Let me know if you have any questions or concerns

Sounds great, will bring my changes there :raised_hands:

**Pull request merged**